lib/validate.go

```diff
@@ -145,12 +165,50 @@ func ValidateQuotaPlan(val *QuotaPlan) error {
 }
 
 // ValidateSurveyURL ...
-func ValidateSurveyURL(val string) error {
-	// yes := govalidator.IsURL(val)
-	// if yes {
-	// 	return nil
-	// }
-	// return ErrInvalidFieldValue
+func ValidateSurveyURL(baseURL string) error {
+	if baseURL == "" {
+		return ErrURLBlank
+	}
+
+	c := utf8.RuneCountInString(baseURL)
+	if c >= URLMaxLength {
+		return ErrURLMaxLength
+	}
+
+	if c <= URLMinLength {
+		return ErrURLMinLength
+	}
+
+	if strings.HasPrefix(baseURL, ".") {
+		return ErrURLPrefix
+	}
+
+	strTemp := baseURL
+	if strings.Contains(baseURL, ":") && !strings.Contains(baseURL, "://") {
+		// support no indicated urlscheme but with colon for port number
+		// http:// is appended so url.Parse will succeed, strTemp used so it does not impact validator.IsURL()
+		strTemp = "http://" + baseURL
+	}
+	u, err := url.Parse(strTemp)
+	if err != nil {
+		return err
+	}
+
+	if strings.HasPrefix(u.Host, ".") {
+		return ErrURLHostPrefix
+	}
+
+	if u.Host == "" && (u.Path != "" && !strings.Contains(u.Path, ".")) {
+		return ErrURLHost
+	}
+
+	if u.Fragment != "" || strings.Contains(baseURL, "#") {
+		return ErrURLFragment
+	}
+
+	if !govalidator.IsURL(baseURL) {
```

Comment:
Did we test this?
We commented this out because our placeholders were causing the validation to fail all the time. lets verify that this works.